src/creosote/cli.py

```diff
@@ -155,23 +144,49 @@ def parse_args(args):
         metavar="FEATURE",
         action="append",
         choices=[v.value for v in Features.__members__.values()],
-        default=[],
         help=(
             "enable new/experimental functionality, "
             "that may be backward incompatible"
         ),
     )
-
-    parsed_args = parser.parse_args(args)
-
-    return parsed_args
+    parser.set_defaults(**dataclasses.asdict(defaults))
 
 
 def main(args_=None):
-    args = parse_args(args_)
+    init_parser = argparse.ArgumentParser(
+        add_help=False,
+    )
+    # Immediately parse verbosity
+    init_parser.add_argument(
+        "--verbose",
+        dest="verbose",
+        action="store_true",
+        help="increase output verbosity",
+    )
+    init_parser.add_argument(
+        "-f",
+        "--format",
+        dest="format",
+        choices=typing.get_args(Config.__annotations__["format"]),
+        help="output format",
+    )
+    args, _ = init_parser.parse_known_args(args_)
 
+    # Configure logger early
     formatters.configure_logger(verbose=args.verbose, format_=args.format)
 
+    # Now that the logger is configured, configure the rest of the parser.
+    parser = argparse.ArgumentParser(
+        description=(
+            "Prevent bloated virtual environments by identifing installed, "
+            "but unused, dependencies"
+        ),
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        parents=[init_parser],
+    )
+    configure_parser(parser, defaults=load_defaults())
+    args = parser.parse_args(args_)
```

Comment:
Hmmm, okay so this means you can't configure the logging as part of `pyproject.toml`. And this is only because I asked you about logging of the `pyproject.toml` file loading?

This all seems kind of messy to me. Let me think about this. Seems like we could load the `pyproject.toml` later, instead of configuring parts of arguments and logging/formatting early.

Reply:
I pushed a second commit, so to simplify this all a bit. Rather than logging in the `load_defaults` function, I just make sure it can return valuable information and I log this a bit later.